test/parallel/test-http-res-write-end-dont-take-array.js

```diff
@@ -23,41 +23,36 @@
 require('../common');
 const assert = require('assert');
 const http = require('http');
-
-let test = 1;
+const Countdown = require('../common/countdown');
 
 const server = http.createServer(function(req, res) {
   res.writeHead(200, { 'Content-Type': 'text/plain' });
-  if (test === 1) {
-    // write should accept string
-    res.write('string');
-    // write should accept buffer
-    res.write(Buffer.from('asdf'));
+  const countdown = new Countdown(2, () => res.end(Buffer.from('asdf')));
```

Comment:
This will no longer do what it claims to as the countdown is redeclared on each request. It will never go below 1 which means the callback will never fire.

In general, this test would be better if rewritten without Countdown and instead by using `.once('request')` with `common.mustCall` for the first test that would then declare `on('request')` with `common.mustCall` for the 2nd test.

Something like this basically:

```js
server.once('request', common.mustCall((req, res) => {
  server.on('request', common.mustCall((req, res) => {
    res.end(Buffer.from('asdf'));	
  }));
  // 1st test case code here
}));
```

Reply:
@apapirovski : Thanks for the feedback. I've updated the PR with the changes and also shortened the commit message. Kindly review the PR now. Thanks !